ONSdigital/ras-frontstage
# Tidy tokens endpoint

# Motivation and Context
This change allows frontstage to work with both the django-oauth-test and ras-rm-auth-service.  The auth service has been changed to return only a 204 when the credentials are correct so that all the unused hardcoded data that was previously being sent is no longer there to confuse people with the possibility of it being important.

# What has changed
The oauth controller now handles 204 and 200 responses and acts accordingly.  If it's 204, we return an empty dict to not mess with code further down the line.  If it's 200, we act as we always did (taking token data, combining with timestamp and party id, then writing it to redis).

# How to test?
  - Run ras-party and ras-fronstage pointing to the old service (port 8040, user 'ons@ons.gov, password 'password').  I did this by running them locally alongside the other docker services (also had to run response-operations-ui locally too)
 -Enrol a new user and sign it in, you should see in the django docker logs that it hits the tokens endpoint to log in.
- Copy the data for the user you just created from the `public -> credentials_oauthuser` for the user into `auth -> user`.
-  Restart the ras-party and ras-fronstage services, but this time with them pointing to the ras-rm-auth-service (port 8041, user 'admin', pass 'secret').  I ran the ras-rm-auth-service locally too but it's your call.
- Try to log in with that user again.  It should work fine, and you should see it hit the tokens endpoint in the logs for ras-rm-auth-service instead of the django one this time.

# Links
https://trello.com/c/XqLdUYwd/1032-tidy-up-tokens-endpoint-in-ras-rm-auth-service-3

# Screenshots (if appropriate):
